File: juyomo/hw3_hw4_1-pointers/54_merge_str_alt.cpp
// Author: Juyoung Moon
// https://leetcode.com/problems/merge-strings-alternately/

// HOCO Mentoring HW 3
// https://github.com/juyomo/hoco-mentoring

class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string res;

        int longerLength = max(word1.size(), word2.size());

        for (int i = 0; i < longerLength; i++) {
            if (i < word1.size()) {
                res += word1[i];
            }
            if (i < word2.size()) {
                res += word2[i];
            }
        }

        return res;
    }
};

File: juyomo/hw3_hw4_1-pointers/54_merge_str_alt.py
# Author: Juyoung Moon
# https://leetcode.com/problems/merge-strings-alternately/

# HOCO Mentoring HW 3
# https://github.com/juyomo/hoco-mentoring

class Solution(object):
    def mergeAlternately(self, word1, word2):
        word12 = []
        for i in range(len(word1+word2)):
            if i < len(word1):
                word12.append(word1[i])
            if i < len(word2):
                word12.append(word2[i])
        return "".join(word12)
